crates/revm/src/processor.rs

```diff
@@ -31,14 +33,17 @@ pub struct EVMProcessor<'a> {
     chain_spec: Arc<ChainSpec>,
     evm: EVM<StateDBBox<'a, Error>>,
     stack: InspectorStack,
-    receipts: Vec<Vec<Receipt>>,
+    /// The collection of receipts.
+    receipts: Vec<HashMap<usize, Receipt>>,
```

Comment:
would appreciate a comment on why `Vec<HashMap<_>>` and not a `Vec<Vec<_>>` as we do in other places